backend/src/graphql/resolver/UserResolver.ts

```diff
@@ -283,14 +289,16 @@ export class UserResolver {
         logger.info('redeemCode found contributionLink=' + contributionLink)
         if (contributionLink) {
           dbUser.contributionLinkId = contributionLink.id
-          eventRegisterRedeem.contributionId = contributionLink.id
+          // TODO this is so wrong
+          eventRegisterRedeem.involvedContribution = { id: contributionLink.id } as DbContribution
         }
       } else {
         const transactionLink = await DbTransactionLink.findOne({ code: redeemCode })
         logger.info('redeemCode found transactionLink=' + transactionLink)
         if (transactionLink) {
           dbUser.referrerId = transactionLink.userId
-          eventRegisterRedeem.transactionId = transactionLink.id
+          // TODO this is so wrong
+          eventRegisterRedeem.involvedTransaction = { id: transactionLink.id } as DbTransaction
```

Comment:
if it is "so wrong", why did you do it or why not refactor?

Reply:
this is fixed in a later commit